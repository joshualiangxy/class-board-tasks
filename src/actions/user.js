import database from '../firebase/firebase';

export const getUserData = (displayName, studentNum) => ({
  type: 'GET_USER_DATA',
  displayName,
  studentNum
});

export const startGetUserData = () => {
  return (dispatch, getState) => {
    const uid = getState().auth.user.uid;

    return database
      .collection('users')
      .doc(uid)
      .get()
      .then(snapshot => {
        const displayName = snapshot.data().displayName;
        const studentNum = snapshot.data().studentNum;
        dispatch(getUserData(displayName, studentNum));
      });
  };
};

export const removeUserData = () => ({ type: 'REMOVE_USER_DATA' });

export const setUserData = (displayName, studentNum) => ({
  type: 'SET_USER_DATA',
  displayName,
  studentNum
});

export const startSetUserData = (displayName, studentNum) => {
  return (dispatch, getState) => {
    const uid = getState().auth.user.uid;

    return database
      .collection('users')
      .doc(uid)
      .set({ displayName, studentNum })
      .then(() => dispatch(setUserData(displayName, studentNum)));
  };
};

export const newUser = () => ({ type: 'NEW_USER' });
