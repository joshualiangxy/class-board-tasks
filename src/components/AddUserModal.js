import React, { useState } from 'react';
import Modal from 'react-modal';
import firebase from '../firebase/firebase';

// Note: This still works off uid for now. This also does NOT work yet.
const AddUserModal = ({ isOpen, onRequestClose, group }) => {
  const addNewUser = (userEmail, group) => {
    const uid = userEmail; // TODO: Change this when email-uid are linked!

    // Both of these references are DocumentReference
    const userRef = firebase.firestore().collection('users').doc(uid);
    const groupRef = firebase.firestore().collection('groups').doc(group);

    // Gets the data from a user.
    userRef.get().then(doc => {
      const data = doc.data();
      // Write displayName, studentNum, and UID into a new document
      if (typeof data === 'undefined') {
        // Most likely a user that doesn't exist.
        setError('No such user found');
      } else {
        groupRef
          .collection('users')
          .doc(uid) // Create document for this user's UID
          .set({
            studentNum: data.studentNum,
            displayName: data.displayName,
            uid: uid,
            admin: false
          })
          .then(() => {
            setError('User added!');
          })
          .catch(error => {
            setError('Something went wrong.');
          });
      }
      // Defaults to non-admin.
    });
  };

  // TODO: When calling this modal, pass in the group UID as a prop
  const [userEmail, setUserEmail] = useState('');
  const [error, setError] = useState('');
  // const [usersAdded, setUsersAdded] = useState([]); // For listing down people that have been added?

  const onUserEmailChange = e => {
    setUserEmail(e.target.value);
  };

  const onCancel = () => {
    setUserEmail('');
    onRequestClose();
  };

  const onSubmit = e => {
    e.preventDefault();
    const submittedEmail = userEmail.trim();
    if (!submittedEmail) {
      setError('Please enter a group name');
    } else {
      addNewUser(submittedEmail, group);
      // TODO: Catch user not found
      // Does not close after submitting, so that the user can continue to add more users

      // setUserEmail(''); // Also resets the field.
      // setError('Added!');
    }
  };

  return (
    <Modal
      isOpen={isOpen}
      contentLabel="New Group"
      onRequestClose={() => onRequestClose()}
      appElement={document.getElementById('root')}
    >
      <form onSubmit={onSubmit}>
        <input
          type="text"
          placeholder="User ID (required)"
          value={userEmail}
          onChange={onUserEmailChange}
          autoFocus
        />
        <button>Submit</button>
        <button onClick={onCancel}>Cancel</button>
        <div>{error}</div>
      </form>
    </Modal>
  );
};

export default AddUserModal;
