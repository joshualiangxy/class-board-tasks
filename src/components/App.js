import React from 'react';
import '../styles/App.css';

const App = () => (
  <div>
    <h1>This is my app</h1>
  </div>
);

export default App;

