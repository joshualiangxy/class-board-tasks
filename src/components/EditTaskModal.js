import React from 'react';
import Modal from 'react-modal';
import { connect } from 'react-redux';
import {
  startEditPersonalTask,
  startEditGroupTask,
  startSetTasks
} from '../actions/tasks';
import TaskForm from './TaskForm';

export const EditTaskModal = ({
  startEditPersonalTask,
  startEditGroupTask,
  startSetTasks,
  isOpen,
  onRequestClose,
  gid,
  groupModule,
  groupName,
  task,
  users,
  editGroupTask
}) => {
  const submitTask = updates =>
    gid
      ? startEditGroupTask(task.id, updates, groupName).then(() => {
          editGroupTask(task.id, updates);
          startSetTasks(); // Temporary solution
          onRequestClose();
        })
      : startEditPersonalTask(task.id, updates).then(() => onRequestClose());

  return (
    <Modal
      isOpen={isOpen}
      contentLabel="Edit Task"
      onRequestClose={onRequestClose}
      appElement={document.getElementById('root')}
    >
      <h2>Edit Task</h2>
      <TaskForm
        gid={gid}
        groupModule={groupModule}
        users={users}
        submitTask={submitTask}
        onRequestClose={onRequestClose}
        task={task}
      />
    </Modal>
  );
};

const mapDispatchToProps = dispatch => ({
  startEditPersonalTask: (id, updates) =>
    dispatch(startEditPersonalTask(id, updates)),
  startEditGroupTask: (id, updates, groupName) =>
    dispatch(startEditGroupTask(id, updates, groupName)),
  startSetTasks: () => dispatch(startSetTasks())
});

export default connect(undefined, mapDispatchToProps)(EditTaskModal);
