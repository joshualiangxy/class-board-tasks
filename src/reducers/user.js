const userReducer = (state = {}, action) => {
  switch (action.type) {
    case 'GET_USER_DATA':
    case 'SET_USER_DATA':
      return {
        displayName: action.displayName,
        studentNum: action.studentNum
      };
    case 'NEW_USER':
      return { newUser: true, ...state };
    case 'REMOVE_USER_DATA':
      return {};
    default:
      return state;
  }
};

export default userReducer;
