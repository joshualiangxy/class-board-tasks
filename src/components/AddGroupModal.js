import React, { useState } from 'react';
import Modal from 'react-modal';
import { startNewGroup } from '../actions/groups';
import { connect } from 'react-redux';

// TODO: Add the ability to add other users into the group
const AddGroupModal = ({ isOpen, onRequestClose, startNewGroup }) => {
  const [groupName, setGroupName] = useState('');
  const [module, setModule] = useState('');
  const [error, setError] = useState('');

  const onGroupNameChange = e => setGroupName(e.target.value);

  const onModuleChange = e => {
    const module = e.target.value.toUpperCase();

    setModule(module);
  };

  const onCancel = () => {
    setError('');
    setGroupName('');
    setModule('');
    onRequestClose();
  };

  const onSubmit = e => {
    e.preventDefault();
    const submittedName = groupName.trim();
    const submittedMod = module.trim();
    if (!submittedName) {
      setError('Please enter a group name');
    } else {
      setError('');
      setGroupName('');
      setModule('');
      startNewGroup(submittedName, submittedMod).then(() => onRequestClose());
    }
  };

  return (
    <Modal
      isOpen={isOpen}
      contentLabel="New Group"
      onRequestClose={onCancel}
      closeTimeoutMS={200}
      appElement={document.getElementById('root')}
    >
      <form onSubmit={onSubmit}>
        <input
          type="text"
          placeholder="Group name (required)"
          value={groupName}
          onChange={onGroupNameChange}
          autoFocus
        />
        <input
          type="text"
          placeholder="Module code/Class name (Optional)"
          value={module}
          onChange={onModuleChange}
        />
        <button>Submit</button>
        <button onClick={onCancel}>Cancel</button>
        <div>{error}</div>
      </form>
    </Modal>
  );
};

const mapDispatchToProps = dispatch => ({
  startNewGroup: (groupName, module) =>
    dispatch(startNewGroup(groupName, module))
});

export default connect(null, mapDispatchToProps)(AddGroupModal);
