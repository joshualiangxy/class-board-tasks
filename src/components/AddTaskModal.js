import React from 'react';
import Modal from 'react-modal';
import { connect } from 'react-redux';
import { startAddPersonalTask, startAddGroupTask } from '../actions/tasks';
import TaskForm from './TaskForm';

export const AddTaskModal = ({
  startAddPersonalTask,
  startAddGroupTask,
  removeAdmin,
  isOpen,
  onRequestClose,
  gid,
  groupModule,
  groupName,
  addGroupTask,
  users
}) => {
  const submitTask = task =>
    gid
      ? startAddGroupTask(task, groupName)
          .then(isAdmin => {
            if (isAdmin) return addGroupTask(task);
            else return removeAdmin();
          })
          .then(() => onRequestClose())
      : startAddPersonalTask(task).then(() => onRequestClose());

  return (
    <Modal
      className="task-modal"
      isOpen={isOpen}
      contentLabel="Add Task"
      onRequestClose={onRequestClose}
      closeTimeoutMS={200}
      appElement={document.getElementById('root')}
    >
      <h1>Add Task</h1>
      <TaskForm
        gid={gid}
        groupModule={groupModule}
        users={users}
        submitTask={submitTask}
        onRequestClose={onRequestClose}
      />
    </Modal>
  );
};

const mapDispatchToProps = dispatch => ({
  startAddPersonalTask: task => dispatch(startAddPersonalTask(task)),
  startAddGroupTask: (task, groupName) =>
    dispatch(startAddGroupTask(task, groupName))
});

export default connect(undefined, mapDispatchToProps)(AddTaskModal);
