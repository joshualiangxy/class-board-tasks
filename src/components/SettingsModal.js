import React, { useState } from 'react';
import Modal from 'react-modal';
import { connect } from 'react-redux';
import { startSetUserData } from '../actions/user';

export const SettingsModal = ({
  initialDisplayName = '',
  initialStudentNum = '',
  isOpen,
  onRequestClose,
  startSetUserData,
  isNewUser
}) => {
  const [displayName, setDisplayName] = useState(initialDisplayName);
  const [studentNum, setStudentNum] = useState(initialStudentNum);
  const [error, setError] = useState('');

  const onDisplayNameChange = e => {
    const displayName = e.target.value;
    setDisplayName(displayName);
  };

  const onStudentNumChange = e => {
    const studentNum = e.target.value;
    if (
      !studentNum ||
      studentNum.match(/^[aA]\d{0,7}$/) ||
      studentNum.match(/^[aA]\d{7}[a-zA-Z]$/)
    )
      setStudentNum(studentNum.toUpperCase());
  };

  const onCancel = () => {
    if (isNewUser) {
      setError('Please submit display name and student number');
      return;
    } else {
      setError('');
      setDisplayName(initialDisplayName);
      setStudentNum(initialStudentNum);
      onRequestClose(false);
    }
  };

  const onSubmit = e => {
    e.preventDefault();

    const submittedDisplayName = displayName.trim();
    if (
      !submittedDisplayName &&
      (!studentNum || !studentNum.match(/^A\d{7}[A-Z]$/))
    ) {
      setError('Please enter display name and student number');
    } else if (!submittedDisplayName) {
      setError('Please enter display name');
    } else if (!studentNum.match(/^A\d{7}[A-Z]$/)) {
      setError('Please enter a valid student number');
    } else {
      setError('');
      onRequestClose(true);
      return startSetUserData(submittedDisplayName, studentNum);
    }
  };

  return (
    <Modal
      className="setting-modal"
      isOpen={isOpen}
      contentLabel="Settings"
      onRequestClose={onCancel}
      appElement={document.getElementById('root')}
      closeTimeoutMS={200}
    >
      {isNewUser ? (
        <div>
          <h1>Welcome to Class Board Tasks!</h1>
          <p>Please enter your information to get started</p>
        </div>
      ) : (
        <h1>Settings</h1>
      )}
      <form className="form" onSubmit={onSubmit}>
        {error && <p className="form__error">{error}</p>}
        <input
          className="text-input"
          type="text"
          placeholder="Display name (required)"
          value={displayName}
          onChange={onDisplayNameChange}
          autoFocus={true}
        />
        <input
          className="text-input"
          type="text"
          placeholder="Student number (required)"
          value={studentNum}
          onChange={onStudentNumChange}
        />
        <div>
          <button className="button button--norm button--right">Submit</button>
          <button
            className="button button--norm"
            type="button"
            onClick={onCancel}
            disabled={isNewUser}
          >
            Cancel
          </button>
        </div>
      </form>
    </Modal>
  );
};

const mapStateToProps = state => ({
  initialDisplayName: state.user.displayName,
  initialStudentNum: state.user.studentNum
});

const mapDispatchToProps = dispatch => ({
  startSetUserData: (displayName, studentNum) =>
    dispatch(startSetUserData(displayName, studentNum))
});

export default connect(mapStateToProps, mapDispatchToProps)(SettingsModal);
