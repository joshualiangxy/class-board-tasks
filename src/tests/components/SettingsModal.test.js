import React from 'react';
import { shallow } from 'enzyme';
import user from '../fixtures/user';
import { SettingsModal } from '../../components/SettingsModal';

const startSetUserData = jest.fn(() => Promise.resolve());
const onRequestClose = jest.fn();
const displayName = user.displayName;
const studentNum = user.studentNum;
let wrapper;

beforeEach(() => {
  wrapper = shallow(
    <SettingsModal
      startSetUserData={startSetUserData}
      onRequestClose={onRequestClose}
    />
  );
  onRequestClose.mockClear();
});

describe('render', () => {
  it('should render SettingsModal', () => expect(wrapper).toMatchSnapshot());

  it('should render SettingsModal with displayName and studentNum', () => {
    wrapper = shallow(
      <SettingsModal
        startSetUserData={startSetUserData}
        initialDisplayName={displayName}
        initialStudentNum={studentNum}
        onRequestClose={onRequestClose}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should render different text and disabled cancel button for new user', () => {
    wrapper.setProps({ isNewUser: true });

    expect(wrapper).toMatchSnapshot();
  });
});

describe('display name input', () => {
  it('should set display name on input change', () => {
    const value = displayName;

    expect(wrapper.find('input').at(0).prop('value')).toBe('');

    wrapper.find('input').at(0).prop('onChange')({ target: { value } });

    expect(wrapper.find('input').at(0).prop('value')).toBe(value);
  });
});

describe('student number input', () => {
  it('should set student number on valid input change', () => {
    const value = studentNum;

    expect(wrapper.find('input').at(1).prop('value')).toBe('');

    wrapper.find('input').at(1).prop('onChange')({ target: { value } });

    expect(wrapper.find('input').at(1).prop('value')).toBe(value);
  });

  it('should set student number to uppercase if lowercase input is valid', () => {
    const value = 'a002';

    expect(wrapper.find('input').at(1).prop('value')).toBe('');

    wrapper.find('input').at(1).prop('onChange')({ target: { value } });

    expect(wrapper.find('input').at(1).prop('value')).toBe(value.toUpperCase());
  });

  it('should not set student number if invalid input', () => {
    const value = 'abc123';

    expect(wrapper.find('input').at(1).prop('value')).toBe('');

    wrapper.find('input').at(1).prop('onChange')({ target: { value } });

    expect(wrapper.find('input').at(1).prop('value')).toBe('');
  });
});

describe('cancel button', () => {
  const newDisplayName = 'robert';
  const newStudentNum = 'A0000000A';

  it('should set display name and student number back to initial state and close modal', () => {
    wrapper.setProps({
      initialDisplayName: displayName,
      initialStudentNum: studentNum
    });
    wrapper.find('input').at(0).prop('onChange')({
      target: { value: newDisplayName }
    });
    wrapper.find('input').at(1).prop('onChange')({
      target: { value: newStudentNum }
    });

    expect(wrapper.find('input').at(0).prop('value')).toBe(newDisplayName);
    expect(wrapper.find('input').at(1).prop('value')).toBe(newStudentNum);

    wrapper.find('button').at(1).prop('onClick')();

    expect(wrapper.find('input').at(0).prop('value')).toBe(displayName);
    expect(wrapper.find('input').at(1).prop('value')).toBe(studentNum);

    expect(onRequestClose).toHaveBeenCalledTimes(1);
    expect(onRequestClose).toHaveBeenLastCalledWith(false);
  });

  it('should render error and not reset input fields if user is new', () => {
    wrapper.setProps({
      isNewUser: true,
      initialDisplayName: displayName,
      initialStudentNum: studentNum
    });
    wrapper.find('input').at(0).prop('onChange')({
      target: { value: newDisplayName }
    });
    wrapper.find('input').at(1).prop('onChange')({
      target: { value: newStudentNum }
    });

    expect(wrapper.find('input').at(0).prop('value')).toBe(newDisplayName);
    expect(wrapper.find('input').at(1).prop('value')).toBe(newStudentNum);

    wrapper.find('button').at(1).prop('onClick')();

    expect(wrapper.find('input').at(0).prop('value')).toBe(newDisplayName);
    expect(wrapper.find('input').at(1).prop('value')).toBe(newStudentNum);

    expect(wrapper).toMatchSnapshot();
  });
});

describe('submit button', () => {
  const submitEvent = { preventDefault: () => {} };

  it('should render error if empty display name and student number on submit', () => {
    wrapper.find('form').prop('onSubmit')(submitEvent);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render error if empty display name and incorrect format for student number', () => {
    wrapper.find('form').prop('onSubmit')(submitEvent);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render error if empty display name', () => {
    wrapper = shallow(
      <SettingsModal
        startSetUserData={startSetUserData}
        initialStudentNum={studentNum}
        onRequestClose={onRequestClose}
      />
    );
    wrapper.find('form').prop('onSubmit')(submitEvent);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render error if invalid student number', () => {
    wrapper = shallow(
      <SettingsModal
        startSetUserData={startSetUserData}
        initialDisplayName={displayName}
        onRequestClose={onRequestClose}
      />
    );
    wrapper.find('form').prop('onSubmit')(submitEvent);

    expect(wrapper).toMatchSnapshot();
  });

  it('should call startSetUserData and onRequestClose on valid submit', () => {
    wrapper = shallow(
      <SettingsModal
        startSetUserData={startSetUserData}
        initialDisplayName={displayName}
        initialStudentNum={studentNum}
        onRequestClose={onRequestClose}
      />
    );
    wrapper
      .find('form')
      .prop('onSubmit')(submitEvent)
      .then(() => {
        expect(wrapper).toMatchSnapshot();

        expect(startSetUserData).toHaveBeenCalledTimes(1);
        expect(startSetUserData).toHaveBeenLastCalledWith(
          displayName,
          studentNum
        );

        expect(onRequestClose).toHaveBeenCalledTimes(1);
        expect(onRequestClose).toHaveBeenLastCalledWith(true);
      });
  });
});
