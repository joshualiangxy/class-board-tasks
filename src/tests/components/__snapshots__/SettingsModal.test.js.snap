// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cancel button should render error and not reset input fields if user is new 1`] = `
<Modal
  appElement={null}
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="setting-modal"
  closeTimeoutMS={200}
  contentLabel="Settings"
  isOpen={false}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <div>
    <h1>
      Welcome to Class Board Tasks!
    </h1>
    <p>
      Please enter your information to get started
    </p>
  </div>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <p
      className="form__error"
    >
      Please submit display name and student number
    </p>
    <input
      autoFocus={true}
      className="text-input"
      onChange={[Function]}
      placeholder="Display name (required)"
      type="text"
      value="robert"
    />
    <input
      className="text-input"
      onChange={[Function]}
      placeholder="Student number (required)"
      type="text"
      value="A0000000A"
    />
    <div>
      <button
        className="button button--norm button--right"
      >
        Submit
      </button>
      <button
        className="button button--norm"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
    </div>
  </form>
</Modal>
`;

exports[`render should render SettingsModal 1`] = `
<Modal
  appElement={null}
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="setting-modal"
  closeTimeoutMS={200}
  contentLabel="Settings"
  isOpen={false}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h1>
    Settings
  </h1>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <input
      autoFocus={true}
      className="text-input"
      onChange={[Function]}
      placeholder="Display name (required)"
      type="text"
      value=""
    />
    <input
      className="text-input"
      onChange={[Function]}
      placeholder="Student number (required)"
      type="text"
      value=""
    />
    <div>
      <button
        className="button button--norm button--right"
      >
        Submit
      </button>
      <button
        className="button button--norm"
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
    </div>
  </form>
</Modal>
`;

exports[`render should render SettingsModal with displayName and studentNum 1`] = `
<Modal
  appElement={null}
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="setting-modal"
  closeTimeoutMS={200}
  contentLabel="Settings"
  isOpen={false}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h1>
    Settings
  </h1>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <input
      autoFocus={true}
      className="text-input"
      onChange={[Function]}
      placeholder="Display name (required)"
      type="text"
      value="michael"
    />
    <input
      className="text-input"
      onChange={[Function]}
      placeholder="Student number (required)"
      type="text"
      value="A0123456B"
    />
    <div>
      <button
        className="button button--norm button--right"
      >
        Submit
      </button>
      <button
        className="button button--norm"
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
    </div>
  </form>
</Modal>
`;

exports[`render should render different text and disabled cancel button for new user 1`] = `
<Modal
  appElement={null}
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="setting-modal"
  closeTimeoutMS={200}
  contentLabel="Settings"
  isOpen={false}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <div>
    <h1>
      Welcome to Class Board Tasks!
    </h1>
    <p>
      Please enter your information to get started
    </p>
  </div>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <input
      autoFocus={true}
      className="text-input"
      onChange={[Function]}
      placeholder="Display name (required)"
      type="text"
      value=""
    />
    <input
      className="text-input"
      onChange={[Function]}
      placeholder="Student number (required)"
      type="text"
      value=""
    />
    <div>
      <button
        className="button button--norm button--right"
      >
        Submit
      </button>
      <button
        className="button button--norm"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
    </div>
  </form>
</Modal>
`;

exports[`submit button should call startSetUserData and onRequestClose on valid submit 1`] = `
<Modal
  appElement={null}
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="setting-modal"
  closeTimeoutMS={200}
  contentLabel="Settings"
  isOpen={false}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h1>
    Settings
  </h1>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <input
      autoFocus={true}
      className="text-input"
      onChange={[Function]}
      placeholder="Display name (required)"
      type="text"
      value="michael"
    />
    <input
      className="text-input"
      onChange={[Function]}
      placeholder="Student number (required)"
      type="text"
      value="A0123456B"
    />
    <div>
      <button
        className="button button--norm button--right"
      >
        Submit
      </button>
      <button
        className="button button--norm"
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
    </div>
  </form>
</Modal>
`;

exports[`submit button should render error if empty display name 1`] = `
<Modal
  appElement={null}
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="setting-modal"
  closeTimeoutMS={200}
  contentLabel="Settings"
  isOpen={false}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h1>
    Settings
  </h1>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <p
      className="form__error"
    >
      Please enter display name
    </p>
    <input
      autoFocus={true}
      className="text-input"
      onChange={[Function]}
      placeholder="Display name (required)"
      type="text"
      value=""
    />
    <input
      className="text-input"
      onChange={[Function]}
      placeholder="Student number (required)"
      type="text"
      value="A0123456B"
    />
    <div>
      <button
        className="button button--norm button--right"
      >
        Submit
      </button>
      <button
        className="button button--norm"
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
    </div>
  </form>
</Modal>
`;

exports[`submit button should render error if empty display name and incorrect format for student number 1`] = `
<Modal
  appElement={null}
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="setting-modal"
  closeTimeoutMS={200}
  contentLabel="Settings"
  isOpen={false}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h1>
    Settings
  </h1>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <p
      className="form__error"
    >
      Please enter display name and student number
    </p>
    <input
      autoFocus={true}
      className="text-input"
      onChange={[Function]}
      placeholder="Display name (required)"
      type="text"
      value=""
    />
    <input
      className="text-input"
      onChange={[Function]}
      placeholder="Student number (required)"
      type="text"
      value=""
    />
    <div>
      <button
        className="button button--norm button--right"
      >
        Submit
      </button>
      <button
        className="button button--norm"
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
    </div>
  </form>
</Modal>
`;

exports[`submit button should render error if empty display name and student number on submit 1`] = `
<Modal
  appElement={null}
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="setting-modal"
  closeTimeoutMS={200}
  contentLabel="Settings"
  isOpen={false}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h1>
    Settings
  </h1>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <p
      className="form__error"
    >
      Please enter display name and student number
    </p>
    <input
      autoFocus={true}
      className="text-input"
      onChange={[Function]}
      placeholder="Display name (required)"
      type="text"
      value=""
    />
    <input
      className="text-input"
      onChange={[Function]}
      placeholder="Student number (required)"
      type="text"
      value=""
    />
    <div>
      <button
        className="button button--norm button--right"
      >
        Submit
      </button>
      <button
        className="button button--norm"
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
    </div>
  </form>
</Modal>
`;

exports[`submit button should render error if invalid student number 1`] = `
<Modal
  appElement={null}
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="setting-modal"
  closeTimeoutMS={200}
  contentLabel="Settings"
  isOpen={false}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h1>
    Settings
  </h1>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <p
      className="form__error"
    >
      Please enter a valid student number
    </p>
    <input
      autoFocus={true}
      className="text-input"
      onChange={[Function]}
      placeholder="Display name (required)"
      type="text"
      value="michael"
    />
    <input
      className="text-input"
      onChange={[Function]}
      placeholder="Student number (required)"
      type="text"
      value=""
    />
    <div>
      <button
        className="button button--norm button--right"
      >
        Submit
      </button>
      <button
        className="button button--norm"
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
    </div>
  </form>
</Modal>
`;
